### ISSUE TITLE
beta / 9.0.2 – Web Refresh : opening a record from the global search results is working only for the entity account

### ISSUE THREAD

**Issue by NicolasPrats** [Microsoft/EasyRepro#291]
Hello,

The following code is not working: 
```c#
using (var xrmBrowser = new Microsoft.Dynamics365.UIAutomation.Api.Browser(options))
{
    xrmBrowser.LoginPage.Login(urlWeb, login, password, null);
                
    xrmBrowser.ThinkTime(500);
    xrmBrowser.Navigation.GlobalSearch("nicolas");

    xrmBrowser.GlobalSearch.OpenRecord("contact", 0);
    xrmBrowser.ThinkTime(4000);
}
```

If we try to run this code, the exception ```Search Results is not available``` is thrown.  It's thrown because no element with ```xpath= id("entityDiv_1")``` were found.  This xpath is only valid when we try to open an account. The good xpath to look for is id("entityDiv_{etc}") where {etc} is the object type code of the entity (1 for account, 2 for contact, …).

Instead of using the object type code, maybe you could locate the div containing the results from the div containing the scrollbar. It’s id seems to be ```scrollbarRegion_{logicalname}```.

**Comment by TYLEROL** [comment]
Thanks @NicolasPrats . I've added this to the backlog and will review it as part of our v9.1 update refresh work.

Regards,
tyler

**Comment by TYLEROL** [comment]
@NicolasPrats - Quick follow-up on this one. I'm testing this in my Dynamics 365 organization on version 9.1.0.1121 and I cannot replicate the problem.

I'm going to tweak the xpath call looking for the search results to be generic and not require the ETC value based on your feedback, and since we filter the results accurately after that via the resultsContainer > entityContainers > entityContainer variable assignments.

I tested this across Account, Contact, Lead, Opportunity, and User entities and had success with each result.

Thanks,
Tyler

**Comment by NicolasPrats** [comment]
Thanks for the update @TYLEROL.

I have tested and I confirm that this "almost" works with contact and with custom entities. The searched records are opened but then the exception "CRM Record is Unavailable or not finished loading. Timeout Exceeded" is thrown. To make the new code works correctly, I had to add a `this.SwitchToContent(); `:

``` C#
records[index].Click(true);
this.SwitchToContent(); //Line added
driver.WaitUntilClickable(By.XPath(Elements.Xpath[Reference.Entity.Form]),
    new TimeSpan(0, 0, 60),
    null,
    d => { throw new Exception("CRM Record is Unavailable or not finished loading. Timeout Exceeded"); }
);
```

This is strange because when I did my tests 15 days ago, this part of the code was working perfectly. Do you have any idea why switching to content is mandatory now ? I have tested on crm version 9.1.0.1467 with Chrome 71.0.3578.98

**Comment by TYLEROL** [comment]
Thanks @NicolasPrats . I tested this a bit further and found I could replicate the same problem on the Users entity. It seems as though the frame/window context in some scenarios was not the expected content.

After reviewing, I added two lines, similar to what you've done - but utilizing what we had implemented under the Grid.OpenRecord() method. We have not had any reports of issues with that scenario.

```
                SwitchToContent();
                driver.WaitForPageToLoad();
```

After additional testing, the results on my end look much better. Please take a look and let me know if you encounter any further issues.

Warm Regards,
Tyler

**Comment by TYLEROL** [comment]
This is now available via the latest releases/v9.1 branch.
